### PR TITLE
Stop using new(T); Start using &T{} #10

### PR DESCRIPTION
**Pull request by karmakaze**
I just ran:
```
for x in `find . -name \*.go`; do sed -i '' -e 's/new(\([^)]*\))/\&\1{}/' $x; done
```

There was one that wasn't changed:
```
pkg/aws/dynamodb/service.go:
-       out := new(GetItemOutput)
+       out := &GetItemOutput{}
```

Results in the error . messge ` invalid pointer type *GetItemOutput for composite literal`.

### CODE REVIEW

File: pkg/aws/dynamodb/service_test.go [hunk 1]
```diff
@@ -33,7 +33,7 @@ func testDynamoDBDynamoPutItem(t *testing.T, cfg *configuration.Configuration) {
 	var input TestDynamoDBDynamoPutItemType
 	input.SomeParam = cfg.DynamoDB.PrimaryKey
 
-	dynamoNewSvc := new(DynamoDB)
+	dynamoNewSvc := &DynamoDB{}
```
Comment (andream16) [inline pkg/aws/dynamodb/service_test.go]: Would you mind also initializing them properly like?

```
dynamoNewSvc := &DynamoDB{
    DynamoDB : dynamoSvc
}
```

File: pkg/aws/dynamodb/session.go [hunk 2]
```diff
@@ -25,7 +25,7 @@ func New(svc *pkgAws.Session, endpoint string) (*DynamoDB, error) {
 		return nil, newSvcErr
 	}
 
-	dynamoSvc := new(DynamoDB)
+	dynamoSvc := &DynamoDB{}
 	dynamoSvc.DynamoDB = dynamodb.New(newSvc)
```
Comment (andream16) [inline pkg/aws/dynamodb/session.go]: Same as above

File: pkg/aws/dynamodb/util_test.go [hunk 3]
```diff
@@ -17,7 +17,7 @@ type TestUnmarshalStreamImageType struct {
 func TestNewPutItemInput(t *testing.T) {
 
 	tableName := "some_name"
-	in := new(TestUnmarshalStreamImageType)
+	in := &TestUnmarshalStreamImageType{}
```
Comment (andream16) [inline pkg/aws/dynamodb/util_test.go]: Same as above.

File: pkg/aws/rekognition/session.go [hunk 4]
```diff
@@ -25,7 +25,7 @@ func New(svc *pkgAws.Session, region string) (*Rekognition, error) {
 		return nil, newSvcErr
 	}
 
-	rekognitionSvc := new(Rekognition)
+	rekognitionSvc := &Rekognition{}
```
Comment (andream16) [inline pkg/aws/rekognition/session.go]: Same as above.

File: pkg/aws/rekognition/util.go [hunk 5]
```diff
@@ -268,7 +268,7 @@ func newInputImage(image []byte) (*rekognition.Image, error) {
 		return nil, intErr.Format(Image, ErrEmptyParameter)
 	}
 
-	out := new(rekognition.Image)
+	out := &rekognition.Image{}
 
 	out.Bytes = image
```
Comment (andream16) [inline pkg/aws/rekognition/util.go]: same as above.

File: pkg/aws/s3/session.go [hunk 6]
```diff
@@ -25,7 +25,7 @@ func New(svc *pkgAws.Session, endpoint string) (*S3, error) {
 		return nil, newSvcErr
 	}
 
-	s3Svc := new(S3)
+	s3Svc := &S3{}
```
Comment (andream16) [inline pkg/aws/s3/session.go]: Same as above.

File: pkg/aws/session.go [hunk 7]
```diff
@@ -19,15 +19,15 @@ func New(input *SessionInput) (*Session, error) {
 		return nil, errors.New(ErrNoRegionProvided)
 	}
 
-	cfg := new(aws.Config)
+	cfg := &aws.Config{}
 	cfg.Region = aws.String(input.region)
```
Comment (andream16) [inline pkg/aws/session.go]: Same as above.

File: pkg/aws/sns/session.go [hunk 8]
```diff
@@ -25,7 +25,7 @@ func New(svc *pkgAws.Session, endpoint string) (*SNS, error) {
 		return nil, newSvcErr
 	}
 
-	snsSvc := new(SNS)
+	snsSvc := &SNS{}
```
Comment (andream16) [inline pkg/aws/sns/session.go]: Same as above.

File: pkg/aws/sqs/session.go [hunk 9]
```diff
@@ -25,7 +25,7 @@ func New(svc *pkgAws.Session, endpoint string) (*SQS, error) {
 		return nil, newSvcErr
 	}
 
-	sqsSvc := new(SQS)
+	sqsSvc := &SQS{}
```
Comment (andream16) [inline pkg/aws/sqs/session.go]: Same as above.

File: pkg/aws/util.go [hunk 10]
```diff
@@ -14,7 +14,7 @@ func NewSessionInput(region string) (*SessionInput, error) {
 		return nil, errors.New(ErrNoRegionProvided)
 	}
 
-	svc := new(SessionInput)
+	svc := &SessionInput{}
```
Comment (andream16) [inline pkg/aws/util.go]: Same as above.

**Comment by karmakaze** [comment]
@AndreaM16 Updated with struct member initializers as noted.

**Comment by andream16** [comment]
Thanks for your time @karmakaze . Great job!

**Pull request merged**